xbmc/windowing/egl/EGLNativeTypeIMX.cpp

```diff
@@ -338,7 +365,11 @@ bool CEGLNativeTypeIMX::ModeToResolution(std::string mode, RESOLUTION_INFO *res)
   res->iScreen       = 0;
   res->bFullScreen   = true;
   res->iSubtitles    = (int)(0.965 * res->iHeight);
-  res->fPixelRatio   = 1.0f;
+
+  // recalculated PR relatively to startup desktop resolution will result in 1.0f
+  // for all res with same AR and allows use of others without need to calibrate before
+  // by user.
+  res->fPixelRatio   = (float)m_init.fPixelRatio / (res->iScreenWidth/res->iScreenHeight);
```

Comment:
@koying 

no display is EDID all zeroes and we end up with insane H/W.
invalid EDID (in terms of bad checksum for instance) ends up with the above too (due to mxc_* probe and init control).

case of structure wise EDID correct but with stupid actual values - we could inject edid, but I have not tested (unfortunately my devices provide correct EDIDs).

At this particular soft dependency (on EDID) I think the actual system state (booted and showing screen) is providing base guarantee of EDID as mxc_hdmi requires it's correct parsing - in CEA mode and in DVI too. Particular H and W bytes then we check again. 

The code will start also with no display & no EDID (as said before), but there are some other mxc_hdmi problems most probably making such session unusable (@fritsch has details if you are interested).